server/saved_objects/saved_objects_wrapper.ts

```diff
@@ -0,0 +1,182 @@
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import _ from 'lodash';
+import {
+  HttpServiceStart,
+  SavedObject,
+  SavedObjectsBaseOptions,
+  SavedObjectsBulkCreateObject,
+  SavedObjectsBulkGetObject,
+  SavedObjectsBulkResponse,
+  SavedObjectsBulkUpdateObject,
+  SavedObjectsBulkUpdateOptions,
+  SavedObjectsBulkUpdateResponse,
+  SavedObjectsCheckConflictsObject,
+  SavedObjectsCheckConflictsResponse,
+  SavedObjectsClientWrapperFactory,
+  SavedObjectsCreateOptions,
+  SavedObjectsDeleteOptions,
+  SavedObjectsFindOptions,
+  SavedObjectsFindResponse,
+  SavedObjectsUpdateOptions,
+  SavedObjectsUpdateResponse,
+} from 'opensearch-dashboards/server';
+import { OpenSearchDashboardsAuthState } from '../auth/types/authentication_type';
+
+export class SecuritySavedObjectsClientWrapper {
+  public httpStart?: HttpServiceStart;
+
+  constructor() {}
+
+  public wrapperFactory: SavedObjectsClientWrapperFactory = (wrapperOptions) => {
+    const state: OpenSearchDashboardsAuthState =
+      (this.httpStart!.auth.get(wrapperOptions.request).state as OpenSearchDashboardsAuthState) ||
+      {};
+
+    const selectedTenant = state.selectedTenant;
+    const username = state.authInfo?.user_name;
+    const globalTenant = '';
+    const defaultTenant = 'default';
+    const privateTenant = '__user__';
+
+    let namespaceValue = selectedTenant;
+
+    const createWithNamespace = async <T = unknown>(
+      type: string,
+      attributes: T,
+      options?: SavedObjectsCreateOptions
+    ) => {
+      if (selectedTenant !== undefined && isPrivateTenant(selectedTenant)) {
+        namespaceValue = selectedTenant + username;
+      }
+      _.assign(options, { namespace: [namespaceValue] });
+      return await wrapperOptions.client.create(type, attributes, options);
+    };
+
+    const bulkGetWithNamespace = async <T = unknown>(
+      objects: SavedObjectsBulkGetObject[] = [],
+      options: SavedObjectsBaseOptions = {}
+    ): Promise<SavedObjectsBulkResponse<T>> => {
+      if (selectedTenant !== undefined && isPrivateTenant(selectedTenant)) {
+        namespaceValue = selectedTenant + username;
+      }
+      _.assign(options, { namespace: [namespaceValue] });
+      return await wrapperOptions.client.bulkGet(objects, options);
+    };
+
+    const findWithNamespace = async <T = unknown>(
+      options: SavedObjectsFindOptions
+    ): Promise<SavedObjectsFindResponse<T>> => {
+      const tenants = state.authInfo?.tenants;
+      const availableTenantNames = Object.keys(tenants!);
+      availableTenantNames.push(defaultTenant);
+      availableTenantNames.push(globalTenant);
+      availableTenantNames.push(privateTenant + state.authInfo?.user_name);
+      _.assign(options, { namespaces: availableTenantNames });
```

Comment:
you may need to check if global and private tenant are enabled.

If  remember correctly, the global and private tenant are included in the authInfo as `global_tenant` and `${username}`, you may want to remove their keys in the `availableTenantNames` and then add corresponding symbols

Reply:
done